Add tdastro

hombit opened the pull request:
@athornton proposed to add it in this tdastro issue: https://github.com/lincc-frameworks/tdastro/issues/378

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

athornton commented:
I'm going to close this PR, and will work on getting tdastro into rubin-env-rsp once rubin-env 11 is out, possibly delaying until linux_arm64 support is there.

hombit commented:
@athornton, thanks. I'm not a maintainer of `sncosmo` (or `sncosmo-feedstock`), but I can look into ARM64 support and make a PR to `sncosmo-feedstock`. Do you need my help?

hombit commented:
@athornton we made `sncosmo` optional and removed it from the conda dependencies in `tdastro` v0.0.6. I updated this PR with that version. Should we reopen this PR now?
